inference-engine/src/transformations/src/transformations/op_conversions/convert_gather_downgrade.cpp

```diff
@@ -39,3 +41,29 @@ ngraph::pass::ConvertGather7ToGather1::ConvertGather7ToGather1() {
     auto m = std::make_shared<pattern::Matcher>(gather_v7, matcher_name);
     register_matcher(m, callback);
 }
+
+ngraph::pass::ConvertGather8ToGather7::ConvertGather8ToGather7() {
+    MATCHER_SCOPE(ConvertGather8ToGather7);
+
+    auto gather_v8 = pattern::wrap_type<ngraph::opset8::Gather>();
+
+    ngraph::matcher_pass_callback callback = [=](pattern::Matcher& m) {
```

Comment:
What if Gather v8 is incompatible with Gather v7?

Reply:
v8 is incompatible only if it receives negative indices, which we unfortunately cannot check on the ngraph stage (except if indices are const). But we need somehow to run models where in IR `version='opset8'` for plugin which do not support v8 (or until they do not support v8).

Can we discuss this personally? since topic is complicated

Reply:
will merge into a single file

Reply:
1. Currently meaningful negative indices (that is intended to select from the end) are used in a single Tacatron2 model and for such cases @achetver have written a transformation https://github.com/openvinotoolkit/openvino/pull/6028 which works before v8->v7. So in this case v8->v7 will cause no harm.

2. 'Hacky' negative indices (that is intended to fill corresponding output with zeros) are used at least in models text-spotting-0003, instance-segmentation-0002 and maybe some other. In this cases situation in inverse **if we do not downgrade gather-8 to gather-7 this will cause problems** since such models are intentionally written 

But when v8 will be enabled by a plugin, in order to solve case 2 I have written a transformation https://github.com/openvinotoolkit/openvino/pull/6474 which will be turned on before nG Function reaches the plugin.

We discussed this solution with plugins. @lazarevevgeny could you please confirm?

Reply:
postponed to implement further in a separate PR

Reply:
Evgeny confirmed by approving this PR